src/jacobians.jl

```diff
@@ -120,7 +120,7 @@ end
 
 
 function _make_Ji(::AbstractArray, xtype, dx, color_i, nrows, ncols)
-    Ji = mapreduce(i -> i==color_i ? dx : zeros(xtype,nrows), hcat, 1:ncols)
+    Ji = mapreduce(i -> i==color_i ? dx : convert(typeof(dx), zeros(xtype,nrows)), hcat, 1:ncols)
```

Comment:
should this instead change the `zeros` call?

Reply:
Well it would have to be a `CuArrays.zeros` instead ... because it needs to produce a CuArray in this case. Once other option is to change `xtype`, which is currently initialized as `eltype(::CuArray)`, which is going to return `Float32`

Reply:
To make this initialization statement generic, I guess we need to use `fill!`

Reply:
or just `zero(dx)`?

Reply:
yes, just `zero(dx)` and append. That would fix all of this.